### PR TITLE
feat: graphiql editor

### LINKED ISSUE #209: feature: Embed GraphiQL on the website
**Technical Requirements**

- Embed GraphiQL on the `/playground` URL
- The UI should keep the top navigation bar
- Ensure the default font family is `Space Grotesk` and the monospace font family is `Space Mono`
- Create a dir called `playground` with all the playground-related components
- Ensure code is modular, understandable, and easy to refactor
- Use typescript everywhere.
- The URL should be configurable. 
	- A text box could be provided to submit the URL. 
	- The URL should also be configurable via Query Params so that tailcall CLI can be integrated.

### PR DESCRIPTION
**Pull request by mehulmathur16**
Supported working videos are present on this comment : https://github.com/tailcallhq/tailcallhq.github.io/pull/210#issuecomment-2067600810

/claim #209
/fixes #209

### CODE REVIEW

File: docusaurus.config.ts [hunk 1]
```diff
@@ -93,6 +93,7 @@ export default {
         // {to: "/enterprise", label: "Enterprise", position: "left"},
         {to: "/docs", label: "Docs", position: "left"},
         {to: "https://blog.tailcall.run/", label: "Blog", position: "left"},
+        {to: "/playground", label: "Playground", position: "left"},
```
Comment (tusharmath) [inline docusaurus.config.ts]: not required.

**Comment by mehulmathur16** [comment]
@tusharmath @amitksingh1490 Adding supporting functionality videos.

Working :


https://github.com/tailcallhq/tailcallhq.github.io/assets/64700961/35e260f6-9c82-4c46-9ff6-248d254994ba

Responsiveness :


https://github.com/tailcallhq/tailcallhq.github.io/assets/64700961/d0c165d9-4a89-4b5b-9e6e-f897524c6036

**Comment by mehulmathur16** [comment]
Working with localhost tailcall server:


https://github.com/tailcallhq/tailcallhq.github.io/assets/64700961/d04eda5b-ade3-4445-a6ad-d3527a51c314

File: src/components/playground/Playground.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      return () => {
+        clearTimeout(handler);
+      };
+    }, [inputValue, delay]);
+  
+    return debouncedValue;
+};
+
+const Playground: React.FC<PlaygroundProps> = ({defaultApiEndpoint}) => {
+    const initialApiEndpoint = new URLSearchParams(window.location.search).get('apiEndpoint') || defaultApiEndpoint || "https://graphql-pokemon2.vercel.app/";
```
Comment (amitksingh1490) [inline src/components/playground/Playground.tsx]:
```suggestion
    const initialApiEndpoint = new URLSearchParams(window.location.search).get('apiEndpoint') || defaultApiEndpoint || "http://localhost:8000";
```

File: src/components/playground/Playground.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    return (
+        <div className='mt-SPACE_06'>
+            <div className='flex px-SPACE_04'>
+                <input
+                name="api-endpoint"
+                type="url"
+                value={inputValue}
+                onChange={(e) => setInputValue(e.target.value)}
+                className={`border border-solid border-tailCall-border-light-500 rounded-lg font-space-grotesk h-11 w-[100%]
```
Comment (amitksingh1490) [inline src/components/playground/Playground.tsx]: use tailwind css

File: src/pages/playground.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import Layout from "@theme/Layout"
+import "graphiql/graphiql.css";
+import Playground from "../components/playground/Playground";
+
+const PlaygroundPage = () => {
+    return (
+        <Layout title="GraphQL Playground" description="Explore, test, and interact with your GraphQL APIs effortlessly">
```
Comment (amitksingh1490) [inline src/pages/playground.tsx]:
```suggestion
        <Layout title="Tailcall Playground" description="Explore, test, and interact with your GraphQL APIs effortlessly">
```

File: src/css/custom.css [hunk 5]
```diff
@@ -556,3 +556,35 @@ nav.menu {
...
+.graphiql-main {
+  height: 80vh;
+}
+
+.graphiql-main * {
+  font-family: 'Space Grotesk', sans-serif !important;
+}
+
+.graphiql-dialog {
+  top: 60% !important;
```
Comment (amitksingh1490) [inline src/css/custom.css:569]: avoid !important

File: src/css/custom.css [hunk 6]
```diff
@@ -556,3 +556,35 @@ nav.menu {
     box-shadow: none;
   }
 }
+
+.graphiql-main {
+  height: 80vh;
+}
+
+.graphiql-main * {
+  font-family: 'Space Grotesk', sans-serif !important;
```
Comment (amitksingh1490) [inline src/css/custom.css]: avoid !important

File: src/css/custom.css (hunk 5 above)
Comment (mehulmathur16) [reply src/css/custom.css:569]: This is used here to override parent graphiql styles.

**Review by amitksingh1490 — APPROVED** [review]

**Pull request merged**